controllers/fenceagentsremediation_controller.go

```diff
@@ -145,37 +172,133 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		r.Log.Error(err, "Invalid sharedParameters/nodeParameters from CR - edit/recreate the CR", "CR's Name", req.Name)
 		return emptyResult, nil
 	}
-	// Add medik8s remediation taint
-	r.Log.Info("Add Medik8s remediation taint", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
+	// Add FAR (medik8s) remediation taint
+	r.Log.Info("Try adding FAR (Medik8s) remediation taint", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
 	if err := utils.AppendTaint(r.Client, far.Name); err != nil {
 		return emptyResult, err
 	}
 	cmd := append([]string{far.Spec.Agent}, faParams...)
 	// The Fence Agent is excutable and the parameters structure are valid, but we don't check their values
-	r.Log.Info("Execute the fence agent", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
-	outputRes, outputErr, err := r.Executor.Execute(pod, cmd)
-	if err != nil {
-		// response was a failure message
-		r.Log.Error(err, "Fence Agent response was a failure", "CR's Name", req.Name)
-		return emptyResult, err
-	}
-	if outputErr != "" || outputRes != SuccessFAResponse+"\n" {
-		// response wasn't failure or sucesss message
-		err := fmt.Errorf("unknown fence agent response - expecting `%s` response, but we received `%s`", SuccessFAResponse, outputRes)
-		r.Log.Error(err, "Fence Agent response wasn't a success message", "CR's Name", req.Name)
-		return emptyResult, err
+
+	if meta.IsStatusConditionPresentAndEqual(far.Status.Conditions, commonConditions.ProcessingType, metav1.ConditionTrue) &&
```

Comment:
>a better check for the FASucceeded condition would be != True. 

👍🏻 
>I think the code which prepares command execution can be moved from above inside this if block? (getting the pod, building params..

I though of that, and I went with the current implementation in order to limit the code sections which are affected by the status conditions. Having said that do you still see a greater value of adding the suggested code under the if block?

Reply:
>Then you can set the condition to False in case execution failed.

ATM there is no ProcessingChangeReason for this use case. But I might add something for that

Reply:
> Having said that do you still see a greater value of adding the suggested code under the if block?

yes, why executing all that code when it's not used 🤷🏼‍♂️ 

> ATM there is no ProcessingChangeReason

tbh, I dislike this "one reason for updating all conditions" pattern anyway, and this is why...